runtime/moonbase/src/lib.rs

```diff
@@ -1281,31 +1283,18 @@ moonbeam_runtime_common::impl_runtime_apis_plus_common! {
 						}
 					};
 
-					// Calculate the fee that will be taken by pallet transaction payment
-					let fee: u64 = TransactionPayment::compute_fee(
-						xt.encode().len() as u32,
-						&dispatch_info,
-						tip,
-					).saturated_into();
-
-					// Calculate how much gas this effectively uses according to the existing mapping
 					let effective_gas =
```

Comment:
Just a hunch, but I feel that ethereum transactions usually provide more gas than needed, which is ultimately refunded. Thus ethereum prioritizes txs solely on `fee-per-gas` and is able to be fair withing its sphere. Pay more, get more.
Substrate txs on the other hand try to estimate the weight as closely as possible, and do not have the notion of `fee-per-gas`, thereby tips being the only functions of (relative) priority in the substrate tx pool. 

I feel that the problem at hand has a dual nature. A priority is meant to yield the following two priorities (in my understanding): 
1. Attempt to fill the block to maximum efficiency (maximize block author's incentive)
2. Incentivize  txs when the sender pays more

[1.] Is a harder problem to solve. Since most eth txs would have their weight over-estimated, it's almost impossible to tell upfront how much of a block they'll fill (or how much gas they'll use). This can cause issues that if a substrate tx is (rightly so) prioritized to be higher than some eth txs (owing to increased network congestion), it'll usually be in front of the line for good reasons. However, the eth txs could then be rejected by the pool since they're mostly overestimated in gas/weight and that overestimated weight might not fit in the block. This will most definitely lead to a bias.

[2.] Is a problem with atleast some wiggle room, since we need to provide a comparison mapping from ethereum's `fee-per-gas` + `tips` to substrate's `tips`. It's not possible to have something like ethereum since users cannot opt to increase the `transaction-multiplier` for substrate txs, effectively proving the intent to pay more. 
A fair comparison that I can think of would be the `additional-fee-per-gas` + `tip` from eth to demarcate intent to pay. This assumes the existence of a `base-fee-per-gas` & the user provided `fee-per-gas`. Converted to currency this can potentially provide a better mapping towards substrate's tips, and subsequently lead to a fairer comparison based on the _intent to pay more_.

Note that, that to fix [1] we actually need to pay attention to not just the increased intent to pay but also the absolute values of the weight/gas to achieve a good block fill ration + incentive the block author. I'd envision using the formulae we use to calculate both eth & substrate fee (so we also override priority of eth txs, in addition to substrate ones) but owing to the overestimated nature of eth txs, it will never be accurate enough.

Reply:
I wonder if we can use the following:

```
Ep = (Eg - Mg + Et)
Sp = St

where,
Ep: Ethereum tx priority
  Eg: Tx gas price, as set by sender
  Mg: Minimum gas price, as determined by the multiplier
  Et: Tx tip

Sp: Substrate tx priority
  St: Tx tip
```

This can potentially help equate the intent of the senders to their readiness to pay more

Reply:
I'm dusting off this PR with a fresh mind. I have a couple other thoughts on the "Eth txns are usually overpriced" problem.

1. Your solution would help this, but it doesn't do anything good for the author-incentive side of the problem. Authors just want to maximize their tip and minimize their work, crediting txns in any way for a gas price they won't end up paying means nothing to authors.
2. It might be worth trying to quantify the downside of this problem to understand what solutions make sense.
3. A really simple band-aid would be to measure some avg of how much overstated Eth `gas_limit - gas_used` tends to be, and apply this in favor of Substrate txns.

Reply:
Resolving this for now, feel free to comment and re-open.